### PR TITLE
[6.3.x] BZ1313125: [GSS](6.2.z) Unauthorized (401) when deploying Artifact with encrypted passwords

### PR DESCRIPTION
**Pull request by manstis**
See https://bugzilla.redhat.com/show_bug.cgi?id=1313125

This is a request from GSS to backport https://issues.jboss.org/browse/GUVNOR-2279 to 6.3.x.

(cherry picked from commit c0c2b4db2b5d8a582be8cc6d948a504520fac65b)

### CODE REVIEW

**Comment by manstis** [comment]
See https://github.com/droolsjbpm/drools/pull/670 also.

File: guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GuvnorM2RepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2015 JBoss Inc
```
Comment (Rikkola) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GuvnorM2RepositoryTest.java]: Well, you know.

File: guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GuvnorM2RepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.*;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Aether.class)
+@PowerMockIgnore({ "javax.crypto.*" })
+public class GuvnorM2RepositoryTest {
```
Comment (Rikkola) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GuvnorM2RepositoryTest.java]: Nice to see PowerMock in use. I have thought about using it couple of times, but worked my way around it.

**Comment by Rikkola** [comment]
+1

File: guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GuvnorM2RepositoryTest.java (hunk 1 above)
Comment (manstis) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GuvnorM2RepositoryTest.java]: It's a cherry-pick - I wasn't going to go through and modify the dates... unless you insist!

Comment (Rikkola) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GuvnorM2RepositoryTest.java]: Won't. Not sure how much they matter anyway when moving files or cherry-picking.

**Comment by csadilek** [comment]
Waiting for the linked PR to be merged first here.

**Comment by psiroky** [comment]
I've just merged the linked one https://github.com/droolsjbpm/drools/pull/670.

**Comment by csadilek** [comment]
6.3.x: 1ae36694762f263b48c93cf7300707121f6f9573

**Pull request closed without merging**